Description: Ugly work-around for GCC 4.9 bug
 Details are in: https://bugs.debian.org/752733
 Upstream GCC PR: http://gcc.gnu.org/PR61214
 .
 wxwidgets3.0 (3.0.1-2) unstable; urgency=medium
 .
   * New patch fno-visibility-inlines-hidden.patch to work around GCC 4.9 bug.
Author: Olly Betts <olly@survex.com>
Forwarded: no
Last-Update: 2014-06-28

--- a/include/wx/event.h
+++ b/include/wx/event.h
@@ -4498,4 +4498,57 @@
 #define wxEVT_COMMAND_TEXT_PASTE              wxEVT_TEXT_PASTE
 #define wxEVT_COMMAND_TEXT_UPDATED            wxEVT_TEXT
 
+#if defined __GNUC__ && __GNUC__ == 4 && __GNUC_MINOR__ == 9
+#if !defined NO_WX3_GCC49_VIRTUAL_HACK && !defined WXBUILDING
+// Ugly work-around for GCC 4.9 bug: https://bugs.debian.org/752733
+#define WX3_GCC49_VIRTUAL_HACK(C) wxEvent* (C::*debian_wx3_gcc49_pr61214_##C##_hack)() const __attribute__((unused)) = &C::Clone
+
+namespace {
+
+WX3_GCC49_VIRTUAL_HACK(wxIdleEvent);
+WX3_GCC49_VIRTUAL_HACK(wxThreadEvent);
+
+#if wxUSE_GUI
+WX3_GCC49_VIRTUAL_HACK(wxActivateEvent);
+WX3_GCC49_VIRTUAL_HACK(wxChildFocusEvent);
+WX3_GCC49_VIRTUAL_HACK(wxClipboardTextEvent);
+WX3_GCC49_VIRTUAL_HACK(wxCloseEvent);
+WX3_GCC49_VIRTUAL_HACK(wxCommandEvent);
+WX3_GCC49_VIRTUAL_HACK(wxContextMenuEvent);
+WX3_GCC49_VIRTUAL_HACK(wxDisplayChangedEvent);
+WX3_GCC49_VIRTUAL_HACK(wxDropFilesEvent);
+WX3_GCC49_VIRTUAL_HACK(wxEraseEvent);
+WX3_GCC49_VIRTUAL_HACK(wxFocusEvent);
+WX3_GCC49_VIRTUAL_HACK(wxHelpEvent);
+WX3_GCC49_VIRTUAL_HACK(wxIconizeEvent);
+WX3_GCC49_VIRTUAL_HACK(wxInitDialogEvent);
+WX3_GCC49_VIRTUAL_HACK(wxJoystickEvent);
+WX3_GCC49_VIRTUAL_HACK(wxKeyEvent);
+WX3_GCC49_VIRTUAL_HACK(wxMaximizeEvent);
+WX3_GCC49_VIRTUAL_HACK(wxMenuEvent);
+WX3_GCC49_VIRTUAL_HACK(wxMouseCaptureChangedEvent);
+WX3_GCC49_VIRTUAL_HACK(wxMouseCaptureLostEvent);
+WX3_GCC49_VIRTUAL_HACK(wxMouseEvent);
+WX3_GCC49_VIRTUAL_HACK(wxMoveEvent);
+WX3_GCC49_VIRTUAL_HACK(wxNavigationKeyEvent);
+WX3_GCC49_VIRTUAL_HACK(wxNcPaintEvent);
+WX3_GCC49_VIRTUAL_HACK(wxNotifyEvent);
+WX3_GCC49_VIRTUAL_HACK(wxPaintEvent);
+WX3_GCC49_VIRTUAL_HACK(wxPaletteChangedEvent);
+WX3_GCC49_VIRTUAL_HACK(wxQueryNewPaletteEvent);
+WX3_GCC49_VIRTUAL_HACK(wxScrollEvent);
+WX3_GCC49_VIRTUAL_HACK(wxScrollWinEvent);
+WX3_GCC49_VIRTUAL_HACK(wxSetCursorEvent);
+WX3_GCC49_VIRTUAL_HACK(wxShowEvent);
+WX3_GCC49_VIRTUAL_HACK(wxSizeEvent);
+WX3_GCC49_VIRTUAL_HACK(wxSysColourChangedEvent);
+WX3_GCC49_VIRTUAL_HACK(wxUpdateUIEvent);
+WX3_GCC49_VIRTUAL_HACK(wxWindowCreateEvent);
+WX3_GCC49_VIRTUAL_HACK(wxWindowDestroyEvent);
+#endif
+
+}
+#endif
+#endif
+
 #endif // _WX_EVENT_H_
